### PR TITLE
[WIP] Address Brakeman redirect_to and SQL injection warnings

### PR DESCRIPTION
**Pull request by cartothemax**
## References

* Issue #1630

## Objectives

These changes will address the various Brakeman warnings present. 

## Notes

This pull request cleans up the warnings from Brakeman that were being generated around unsafe `redirects` and possible SQL injection vulnerabilities

### CODE REVIEW

**Comment by cartothemax** [comment]
I will address all of the Hound errors once I have gotten the Brakeman warnings/errors/issues taken care of 😄

File: app/views/budgets/investments/_ballot.html.erb [hunk 1]
```diff
@@ -59,7 +59,7 @@
                 verify_account: verify_account, signin: signin,
                 signup: signup, my_heading: my_heading,
                 change_ballot: change_ballot,
-                heading_link: heading_link(@assigned_heading, @budget)).html_safe %>
+                heading_link: sanitize heading_link(@assigned_heading, @budget)).html_safe %>
```
Comment (javierm) [inline app/views/budgets/investments/_ballot.html.erb]: @cartothemax Thanks for having a look at this! Actually I'm preparing a pull request which will remove all usages of `.html_safe`. Sorry I didn't tell you :pray:; I thought you were only going to work on the redirect warnings.

So, if it isn't asking too much, could you remove the commit dealing with XSS issues (with `git rebase`)? That way our pull requests won't have any conflicts :thinking:.

I'll add a link to the pull request when I open it, so you can review it if you'd like to :wink:.

Comment (javierm) [reply app/views/budgets/investments/_ballot.html.erb]: FYI, I've opened pull request #3747, although it still needs some work.

Comment (cartothemax) [reply app/views/budgets/investments/_ballot.html.erb]: No problem! I got a little carried away chasing down all of the Brakeman errors rather then just sticking to the redirect warnings like the issue stated 🤣

**Pull request closed without merging**